packages/flutter_tools/lib/src/version.dart

```diff
@@ -752,24 +752,47 @@ class GitTagVersion {
         _runGit('git fetch $_flutterGit --tags', processUtils, workingDirectory);
       }
     }
-    // `--match` glob must match old version tag `v1.2.3` and new `1.2.3-dev.4.5`
-    return parse(_runGit('git describe --match *.*.* --first-parent --long --tags', processUtils, workingDirectory));
+    final List<String> tags = _runGit(
+      'git tag --contains HEAD', processUtils, workingDirectory).split('\n');
+    // Check first for a stable tag
+    for (final String tag in tags) {
+      final String trimmedTag = tag.trim();
+      if (RegExp(r'^\d+\.\d+\.\d+$').hasMatch(trimmedTag)) {
+        return parse(trimmedTag);
+      }
+    }
+    // Next check for a dev tag
+    for (final String tag in tags) {
+      final String trimmedTag = tag.trim();
+      if (RegExp(r'^\d+\.\d+\.\d+-\d+\.\d+\.pre$').hasMatch(trimmedTag)) {
+        return parse(trimmedTag);
+      }
+    }
+
+    // If we're not currently on a tag, use git describe
```

Comment:
If we're not currently on a tag, use git describe to find the most recent version tag.

Reply:
done